### PR TITLE
Document some string utilities

### PR DESCRIPTION
**Pull request by ravepossum**
Renames and documents `string_util` and `message_util` files.

### CODE REVIEW

File: include/string_util.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#ifndef POKEPLATINUM_STRING_UTIL_H
+#define POKEPLATINUM_STRING_UTIL_H
+
+
```
Comment (lhearachel) [inline include/string_util.h:4]: [nitpick] Extra newline

File: src/message_util.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#include "string_template.h"
+#include "strbuf.h"
+#include "constants/heap.h"
+#include "constants/message_banks.h"
+#include "constants/narc.h"
+
+Strbuf* MessageUtil_ExpandedStrbuf (StringTemplate *template, MessageLoader *loader, u32 entryID, u32 heapID)
+{
+    Strbuf *ret = NULL;
+    Strbuf *buffer = Strbuf_Init(EXPANDED_STRING_SIZE, HEAP_ID_SYSTEM);
```
Comment (lhearachel) [inline src/message_util.c:18]: [nitpick] I'd prefer that this just be called `buf`, for standardization. Or `str`, but preferably `buf`.

File: include/string_util.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#include <nnsys.h>
+
+#define MAX_STRING_COPY_LEN    256
+#define ASCII_LINE_FEED        0xA
+#define ASCII_CARRIAGE_RETURN  0xD
```
Comment (lhearachel) [inline include/string_util.h:9]: [question] Why not just use `'\n'` and `'\r'`?

File: src/string_util.c [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    return length;
+}
+
+char* StringUtil_CopyToTerminator (char *src, char *dst, char terminator)
+{
+    int i;
+
+    for (i = 0; i < MAX_STRING_COPY_LEN; i++) {
```
Comment (lhearachel) [inline src/string_util.c:23]: [polish]
```suggestion
    for (int i = 0; i < MAX_STRING_COPY_LEN; i++) {
```

File: include/string_util.h (hunk 3 above)
Comment (ravepossum) [reply include/string_util.h]: I wish I could say I had a good reason :)
fixed

**Review by lhearachel — APPROVED** [review]

**Review by lhearachel — APPROVED** [review]

**Pull request merged**